ui/src/pages/Admin/AllStories/index.tsx

```diff
@@ -237,62 +338,30 @@ export const AllStories: React.FC = () => {
     event: React.ChangeEvent<Record<string, unknown>>,
     newValue: number
   ) => {
-    cancelSearch();
-    dispatch({ type: "SET_TAB_VALUE", newValue });
-  };
-
-  const requestSearchHelper = (row, searchedVal: string) => {
-    let doesExist = false;
-    Object.keys(row).forEach((prop) => {
-      //Exclude search for StoryView members not displayed on table cells
-      const excludedParameters = prop !== "ID";
-      prop !== "image_url" &&
-        prop !== "video_url" &&
-        prop !== "content" &&
-        prop !== "is_visible";
-      const numExist =
-        typeof row[prop] === "number" &&
-        row[prop].toString().includes(searchedVal) &&
-        excludedParameters;
-      const stringExist =
-        typeof row[prop] === "string" &&
-        row[prop].toLowerCase().includes(searchedVal.toLowerCase()) &&
-        excludedParameters;
-      if (stringExist || numExist) {
-        doesExist = true;
-      }
+    const tagBooleans = {};
+    state.tags.forEach((tag) => {
+      tagBooleans[tag] = false;
     });
-    return doesExist;
+    const newState = {
+      ...state,
+      search: "",
+      filterState: {
+        visibility: {
+          visible: false,
+          nonVisible: false,
+        },
+        tags: tagBooleans,
+      },
+    };
+    dispatch({ type: "SET_TAB_VALUE", newValue, newState });
```

Comment:
again IMO the state transform should be done in the reducer, the action should only provide the variables needed to perform the transform.